pkg/cache/clusterqueue.go

```diff
@@ -462,17 +490,34 @@ func (c *clusterQueue) updateWorkloadUsage(log logr.Logger, wi *workload.Info, m
 		}
 	}
 	if features.Enabled(features.TopologyAwareScheduling) && wi.IsUsingTAS() {
-		for tasFlavor, tasUsage := range wi.TASUsage() {
-			if tasFlvCache := c.tasFlavorCache(tasFlavor); tasFlvCache != nil {
-				if m == 1 {
-					tasFlvCache.addUsage(tasUsage)
-				}
-				if m == -1 {
-					tasFlvCache.removeUsage(tasUsage)
+		key := workload.Key(wi.Obj)
+		log := log.WithValues("workload", key)
+		if c.isTASSynced() {
+			for tasFlavor, tasUsage := range wi.TASUsage() {
+				if tasFlvCache := c.tasCache.Get(tasFlavor); tasFlvCache != nil {
```

Comment:
this could also be a guard clause to reduce nesting/improve readability

```suggestion
                                tasFlvCache := c.tasCache.Get(tasFlavor)
				if tasFlvCache == nil {
				  log.V(2).Info("TAS flavor used by workload not found in cache", "tasFlavor", tasFlavor)
				  continue
				}
```